components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsFeature.kt

```diff
@@ -120,18 +120,23 @@ class SitePermissionsFeature(
 
             if (shouldStore) {
                 ioCoroutineScope.launch {
-                    storeSitePermissions(request, grantedPermissions, ALLOWED)
+                    storeSitePermissions(session, request, grantedPermissions, ALLOWED)
                 }
             }
             true
         }
     }
 
     @Synchronized internal fun storeSitePermissions(
+        session: Session,
         request: PermissionRequest,
         permissions: List<Permission> = request.permissions,
         status: SitePermissions.Status
     ) {
+        if (session.private) {
+            return
+        }
```

Comment:
I wonder if we should do that check one level higher where we also check `shouldStore` .. there's probably no need to launch the coroutine?

Reply:
Yeah, that could be much better. I will add the coroutine launch in `storeSitePermissions` function!